include/freerdp/utils/chunkstream.h

```diff
@@ -0,0 +1,215 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ *
+ * Copyright 2020 Hardening <contact@hardening-consulting.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FREERDP_UTILS_CHUNKSTREAM_H
+#define FREERDP_UTILS_CHUNKSTREAM_H
+
+#include <winpr/wtypes.h>
+#include <winpr/stream.h>
+#include <freerdp/api.h>
+
+typedef struct _ChunkStream ChunkStream;
+typedef struct _ChunkStreamSlot ChunkStreamSlot;
+
+#define CHUNKSTREAM_MAX_SLOTS 50
```

Comment:
Might better be private, in the `.c` file

Reply:
Well, that's what I originally did, but I figured that a caller might be interested to know how many slots/chunks it can create.
could also be a getter on the `ChunkStream` though.